lib/reports/BackupReportManager.js

```diff
@@ -4,85 +4,131 @@ const _ = require('lodash');
 const Promise = require('bluebird');
 const moment = require('moment-timezone');
 const Repository = require('../db').Repository;
+const logger = require('../logger');
 const CONST = require('../constants');
 const config = require('../config');
 
 class BackupReportManager {
   static getInstanceBackupSummary(instanceId, startTime, endTime) {
+    function getJobRunDetailCount(start, end) {
+      const criteria = {
+        type: CONST.JOB.SCHEDULED_BACKUP,
+        'data.instance_id': instanceId,
+        'response.delete_backup_status.instance_deleted': {
+          $ne: true
+        },
+        createdAt: {
+          $gt: start,
+          $lt: end
+        }
+      };
+      return Repository.count(CONST.DB_MODEL.JOB_RUN_DETAIL, criteria);
+    }
+
     return Promise
       .all([
-        this.getBackupTriggerRecord(instanceId, startTime, endTime),
+        this.getBackupTriggerCount(instanceId, startTime, endTime),
         this.getBackupResult(instanceId, startTime, endTime),
         this.getLastRunDetail(instanceId),
-        this.getJobDetails(instanceId)
+        this.getJobDetails(instanceId),
+
       ])
-      .spread((triggeredBackups, resultBackups, lastRunDetail, jobDetail) => {
+      .spread((triggeredBackupCount, resultBackups, lastRunDetail, jobDetail) => {
         const instanceRecord = {};
         const noBackupDays = [];
         //iterator to first entry of the day
         let startIndex = 0;
         let consecutiveBackupFailureCount = 0;
         let currentDay = new Date(startTime);
         let backupSuccessCount = 0;
-        if (jobDetail) {
-          if (startTime < moment.utc(jobDetail.createdAt).startOf('day').toDate()) {
-            currentDay = moment.utc(jobDetail.createdAt).startOf('day').toDate();
-          }
-          _.assign(instanceRecord, {
-            instanceCreateTime: jobDetail.createdAt
-          });
-        } else {
-          currentDay = new Date(endTime);
-        }
-        if (lastRunDetail && (endTime > lastRunDetail.createdAt)) {
-          _.assign(instanceRecord, {
-            instanceDeleteTime: lastRunDetail.createdAt
+        // Check if start time is less than instance create time
+        return Promise.all([
+            Promise.try(() => {
+              if (jobDetail) {
```

Comment:
Method growing too big. Modularize sections within 45-85 into two seperate functions and name the function which highlights that you are getting the correctedStartDate/endDate for the report based on the create/delete times of instance.

Reply:
Done!